cmake/HighFiveConfig.cmake

```diff
@@ -0,0 +1,19 @@
+include(CMakeFindDependencyMacro)
+
+if(NOT DEFINED HIGHFIVE_FIND_HDF5)
+  set(HIGHFIVE_FIND_HDF5 On)
+endif()
+
+if(HIGHFIVE_FIND_HDF5)
+  find_dependency(HDF5)
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
+
+if(HDF5_IS_PARALLEL)
+  find_dependency(MPI)
+  target_link_libraries(HighFive::HighFive INTERFACE MPI::MPI_C MPI::MPI_CXX)
+endif()
+
+add_library(HighFive ALIAS HighFive::HighFive)
+add_library(HighFiveInclude ALIAS HighFive::Include)
```

Comment:
Do we want this? Does this just encourage more confusion between targets and library names?

Reply:
In `CMakeLists.txt` we create a target called `HighFive`. If we export under a namespace it'll be called `HighFive::HighFive` when doing `find_package(HighFive ...)`. However, when we do `add_subdirectory(third_party/HighFive)` it'll still be called `HighFive`.

The idea is to have both mean the same thing with the hope that the difference then doesn't matter. We might be able to change it such that `HighFive::HighFive` is the only target everywhere.

Does the `::` convince CMake to always consider it a target rather than a string that could be converted to `libHighFive.so`; or do we just end up with `libHighFive::HighFive.so`?

Reply:
> Does the :: convince CMake to always consider it a target rather than a string that could be converted to libHighFive.so; or do we just end up with libHighFive::HighFive.so?

I _think_ that plain `HighFive` is ambiguous and may be library name or target, whereas `HighFive::HighFive` is only a target (valid or not).

Don't we alias target `HighFive` to `HighFive::HighFive` elsewhere in the CMake? We can leave this in, just seems like `HighFive::HighFive` would be more "canonical" by now (thanks CMake, for setting clear standards :) )

Reply:
Easy enough to test. It fails with the following error:
```
CMake Error at CMakeLists.txt:5 (target_link_libraries):
  Target "foo" links to:

    HighFive::HighFive

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

> Don't we alias target HighFive to HighFive::HighFive elsewhere in the CMake?
Yes, in `CMakeLists.txt` we naturally have `HighFive` and create an alias `HighFive::HighFive`. While in `*Config.cmake` we naturally have `HighFive::HighFive` and create an alias `HighFive`. This way the two are synonyms and always available.

We could try if we can hide/eliminate the target `HighFive` (and `HighFiveInclude`) in favour of `HighFive::HighFive`.

Reply:
If CMake doesn't hide it, it's exposed :)

I suppose if we already have `HighFive` when used as a git submodule, we should keep it in the CMake install, too,

Reply:
Yes, I came to the same conclusion. Renaming the target doesn't help, because even if there is no target called `HighFive`, the problematic line:
```
target_link_libraries(foo PUBLIC HighFive)
```
remains valid during configuration and only fails at compile time. Therefore, it's probably better to have the target so that if people correctly find or include HighFive they're not punished of guessing that the target is called `HighFive` like is was in `v2`.